Error "[SocketCore.cc:1015] errorCode=1 SSL/TLS handshake failure: hostname does not match"

I'm sure my site has been configured correctly because firefox, curl, wget and aria2c v1.19.0 from apt work well. aria2c v1.34.0 from pacman gave me the error.

@tatsuhiro-t I was using a server on lan,but you can test with openssl(1):
`$mkdir /tmp/test && cd /tmp/test`
`$openssl req -x509 -newkey rsa:2048 -keyout sign.key -out sign.crt -nodes #generate CA certificate`
`$openssl req -new -newkey rsa:2048 -keyout localhost.key -out localhost.req -nodes`
`$openssl x509 -CAcreateserial -CAkey sign.key -CA sign.crt -req -in localhost.req -out localhost.crt #type localhost as Comman Name in interactive interface`
`#openssl s_server -key localhost.key -cert localhost.crt -accept 443 -www  #Use a higher port number without root`
`$curl https://localhost --cacert /tmp/test/sign.crt  #curl works well in Arch Linux`
`$aria2c --no-conf --ca-certificate=/tmp/test/sign.crt  https://localhost #aria2c doesn't work :(`

06/05 17:41:06 [NOTICE] Downloading 1 item(s)

06/05 17:41:06 [ERROR] CUID#7 - Download aborted. URI=https://localhost
Exception: [AbstractCommand.cc:351] errorCode=1 URI=https://localhost
  -> [SocketCore.cc:1015] errorCode=1 SSL/TLS handshake failure: hostname does not match

06/05 17:41:06 [NOTICE] Download GID#bc52623affd5a435 not complete: 

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
bc5262|ERR |       0B/s|https://localhost

Status Legend:
(ERR):error occurred.

Sorry for my poor english.

--- a/src/LibgnutlsTLSSession.cc
+++ b/src/LibgnutlsTLSSession.cc
@@ -341,6 +341,9 @@ int GnuTLSSession::tlsConnect(const std::string& hostname, TLSVersion& version,
             commonName.assign(altName, altNameLen);
           }
         }
+        else {
+          commonName.assign(altName, altNameLen);
+        }
       }
     }
     if (!net::verifyHostname(hostname, dnsNames, ipAddrs, commonName)) {

